File: WidgetDesigners/LeaseContactForm/LeaseContactFormDesigner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using SitefinityWebApp.WidgetDesigners.LeaseContactForm;
using Telerik.Sitefinity.Web.UI.ControlDesign;

namespace SitefinityWebApp.WidgetDesigners.LeaseContactForm
{
    /// <summary>
    /// Represents a designer for the <typeparamref name="SitefinityWebApp.Mvc.Controllers.LeaseContactFormController"/> widget
    /// </summary>
    public class LeaseContactFormDesigner : ControlDesignerBase
    {
        #region Properties
        /// <summary>
        /// Obsolete. Use LayoutTemplatePath instead.
        /// </summary>
        protected override string LayoutTemplateName
        {
            get
            {
                return string.Empty;
            }
        }

        /// <summary>
        /// Gets the layout template's relative or virtual path.
        /// </summary>
        public override string LayoutTemplatePath
        {
            get
            {
                if (string.IsNullOrEmpty(base.LayoutTemplatePath))
                    return LeaseContactFormDesigner.layoutTemplatePath;
                return base.LayoutTemplatePath;
            }
            set
            {
                base.LayoutTemplatePath = value;
            }
        }

        protected override HtmlTextWriterTag TagKey
        {
            get
            {
                return HtmlTextWriterTag.Div;
            }
        }
        #endregion

        #region Control references
        protected virtual Control InternalRecipients
        {
            get
            {
                return this.Container.GetControl<Control>("InternalRecipients", true);
            }
        }

        protected virtual Control EmailSubjectLine
        {
            get
            {
                return this.Container.GetControl<Control>("EmailSubjectLine", true);
            }
        }

        #endregion

        #region Methods
        protected override void InitializeControls(Telerik.Sitefinity.Web.UI.GenericContainer container)
        {
            // Place your initialization logic here
        }
        #endregion

        #region IScriptControl implementation
        /// <summary>
        /// Gets a collection of script descriptors that represent ECMAScript (JavaScript) client components.
        /// </summary>
        public override System.Collections.Generic.IEnumerable<System.Web.UI.ScriptDescriptor> GetScriptDescriptors()
        {
            var scriptDescriptors = new List<ScriptDescriptor>(base.GetScriptDescriptors());
            var descriptor = (ScriptControlDescriptor)scriptDescriptors.Last();

            descriptor.AddElementProperty("internalRecipients", this.InternalRecipients.ClientID);
            descriptor.AddElementProperty("emailSubjectLine", this.EmailSubjectLine.ClientID);


            return scriptDescriptors;
        }

        /// <summary>
        /// Gets a collection of ScriptReference objects that define script resources that the control requires.
        /// </summary>
        public override System.Collections.Generic.IEnumerable<System.Web.UI.ScriptReference> GetScriptReferences()
        {
            var scripts = new List<ScriptReference>(base.GetScriptReferences());
            scripts.Add(new ScriptReference(LeaseContactFormDesigner.scriptReference));
            return scripts;
        }
        #endregion

        #region Private members & constants
        public static readonly string layoutTemplatePath = "~/WidgetDesigners/LeaseContactForm/LeaseContactFormDesigner.ascx";
        public const string scriptReference = "~/WidgetDesigners/LeaseContactForm/LeaseContactFormDesigner.js";
        #endregion
    }
}

File: WidgetDesigners/LeaseContactForm/LeaseContactFormDesigner.js
Type.registerNamespace('SitefinityWebApp.WidgetDesigners.LeaseContactForm');

SitefinityWebApp.WidgetDesigners.LeaseContactForm.LeaseContactFormDesigner = function (element) {
    /* Initialize Message fields */
    this._internalRecipients = null;
    this._emailSubjectLine = null;

    /* Calls the base constructor */
    SitefinityWebApp.WidgetDesigners.LeaseContactForm.LeaseContactFormDesigner.initializeBase(this, [element]);
}

SitefinityWebApp.WidgetDesigners.LeaseContactForm.LeaseContactFormDesigner.prototype = {
    /* --------------------------------- set up and tear down --------------------------------- */
    initialize: function () {
        /* Here you can attach to events or do other initialization */
        SitefinityWebApp.WidgetDesigners.LeaseContactForm.LeaseContactFormDesigner.callBaseMethod(this, 'initialize');
    },
    dispose: function () {
        /* this is the place to unbind/dispose the event handlers created in the initialize method */
        SitefinityWebApp.WidgetDesigners.LeaseContactForm.LeaseContactFormDesigner.callBaseMethod(this, 'dispose');
    },

    /* --------------------------------- public methods ---------------------------------- */

    findElement: function (id) {
        var result = jQuery(this.get_element()).find('#' + id).get(0);
        return result;
    },

    /* Called when the designer window gets opened and here is place to "bind" your designer to the control properties */
    refreshUI: function () {
        var controlData = this._propertyEditor.get_control().Settings; /* JavaScript clone of your control - all the control properties will be properties of the controlData too */

        /* RefreshUI Message */
        jQuery(this.get_internalRecipients()).val(controlData.InternalRecipients);
        jQuery(this.get_emailSubjectLine()).val(controlData.EmailSubjectLine);
    },

    /* Called when the "Save" button is clicked. Here you can transfer the settings from the designer to the control */
    applyChanges: function () {
        var controlData = this._propertyEditor.get_control().Settings;

        /* ApplyChanges Message */
        controlData.InternalRecipients = jQuery(this.get_internalRecipients()).val();
        controlData.EmailSubjectLine = jQuery(this.get_emailSubjectLine()).val();
    },

    /* --------------------------------- event handlers ---------------------------------- */

    /* --------------------------------- private methods --------------------------------- */

    /* --------------------------------- properties -------------------------------------- */

    /* InternalRecipients properties */
    get_internalRecipients: function () { return this._internalRecipients; },
    set_internalRecipients: function (value) { this._internalRecipients = value; },
    /* EmailSubjectLine properties */
    get_emailSubjectLine: function () { return this._emailSubjectLine; },
    set_emailSubjectLine: function (value) { this._emailSubjectLine = value; }
}

SitefinityWebApp.WidgetDesigners.LeaseContactForm.LeaseContactFormDesigner.registerClass('SitefinityWebApp.WidgetDesigners.LeaseContactForm.LeaseContactFormDesigner', Telerik.Sitefinity.Web.UI.ControlDesign.ControlDesignerBase);
